test/e2e/webdriver/driver.js

```diff
@@ -1443,24 +1443,62 @@ class Driver {
   }
 
   /**
-   * Get the text of the alert popup that is currently open in the browser
-   * session.
-   *
-   * @param text - The text of the alert popup.
-   * @param options - Options for the function.
-   * @param options.timeout - The maximum time to wait for the alert to be
-   * present.
-   * @returns {Promise<string>} The text of the alert popup.
+   * Wait for a browser alert to appear and verify its text. Automatically detects
+   * whether a window switch is needed by comparing current window with target window.
+   * If they match, uses normal alert waiting. If different, attempts window switching
+   * but handles the case where Selenium fails to switch to a window with an active alert
+   * (throws UnexpectedAlertOpenError) by extracting the alert text from the error message.
+   *
+   * @param {object} options - Required options for the function.
+   * @param {string} options.text - The expected text of the alert popup.
+   * @param {string} options.windowTitle - The target window title where the alert should appear.
+   * @param {number} [options.timeout] - The maximum time to wait for the alert to be present.
+   * @returns {Promise<void>} Promise that resolves when alert is found and verified.
    */
-  async waitForAlert(text, { timeout = this.timeout } = {}) {
-    await this.driver.wait(until.alertIsPresent(), timeout);
-    const alert = await this.driver.switchTo().alert();
-    const alertText = await alert.getText();
+  async waitForBrowserAlert({ text, windowTitle, timeout = this.timeout }) {
```

Comment:
ℹ️  I've renamed it to make it a bit more meaningful, and to not be confused with other kind of alerts (MM UI specific).
Also, added all the function params in a unique object, for better clarity